src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,123 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * Deals with interactions with the user.
+ */
+public class Ui {
+
+    private Scanner input = new Scanner(System.in);
+
+    /**
+     * Obtains the user input.
+     *
+     * @return String representation of the user input.
+     */
+    public String getInput() {
+        return input.nextLine();
+    }
+
+    /**
+     * Greets the user when the chatbot is started.
+     */
+    public void printGreeting() {
+        String greet = "Hi! I'm Uke\n" + "What can I do for you?\n";
+        System.out.println(greet);
+    }
+
+    /**
+     * Prints the list of tasks in the TaskList given.
+     *
+     * @param storedTasks TaskList to be printed.
+     * @param isListCommand Whether the method is called from a list command.
+     */
+    public void printList(TaskList storedTasks, boolean isListCommand) {
+        int len = storedTasks.getLength();
+
+        if (isListCommand) {
+            if (len > 0) {
+                System.out.println("Okay! Here's the full list of your added tasks:");
+                for (int i = 1; i < len + 1; i++) {
+                    System.out.println(i + ". " + storedTasks.getTask(i - 1));
+                }
+            } else {
+                System.out.println("No tasks found!");
+            }
+        } else {
+            if (len > 0) {
+                System.out.println("Here's the list of matching tasks I found:");
+                for (int i = 1; i < len + 1; i++) {
+                    System.out.println(i + ". " + storedTasks.getTask(i - 1));
+                }
+            } else {
+                System.out.println("No matching tasks found :(");
```

Comment:
Perhaps can add another String variable to store a string value conditionally if this error message needs to be printed out? 🤔 
For example:
```java
String emptyTasksMessage = isListCommand
        ? "No tasks found!"
        : "No matching tasks found :(";
```